Modules/_csv.c

```diff
@@ -1213,6 +1213,19 @@ csv_writerow(WriterObj *self, PyObject *seq)
             append_ok = join_append(self, field, quoted);
             Py_DECREF(field);
         }
+        else if (PyBool_Check(field)) {
+            PyObject *str;
+            double d = PyFloat_AsDouble(field);
+            str = PyFloat_FromDouble(d);
```

Comment:
```
            str = PyObject_Str(str);
            Py_DECREF(str);
```
You have made even worse mistake. Not just the float object is leaked, but the str object is decrefed twice. Use different variables for saving results of `PyFloat_FromDouble()` and `PyObject_Str()`.